File: activities and workings/java/src/extra2Exercise/audioControlExercise/typescript/AudioControl.ts

export class AudioControl{

    // attributes
    private volumeValue : number;

    // constructor
    constructor() {
        this.volumeValue = 2;   // default
    }

    // methods
    increaseVolume() : void {

        if (this.volumeValue == 10){
            console.log("\n> volume is already at maximum: " + this.volumeValue);
            return;
        }

        this.volumeValue++;
    }

    decreaseVolume() : void {

        if (this.volumeValue == 1){
            console.log("volume is already at minimum: " + this.volumeValue);
            return;
        }

        this.volumeValue--;
    }

    readVolume() : number {
        return this.volumeValue;
    }

}

function main(){

    const v : AudioControl = new AudioControl();

    v.increaseVolume();
    v.increaseVolume();
    v.increaseVolume();
    v.decreaseVolume();

    console.log("current volume: " + v.readVolume());   // out: 4

}

main();

File: atividades e trabalhos/java/src/extra2Exercise/audioControlExercise/java/AudioControl.java
package extra2Exercise.audioControlExercise.java;

public class AudioControl {

    // attributes
    private Integer volumeValue;

    // constructor
    public AudioControl(){
        this.volumeValue = 2;   // default
    }

    // methods

    public void increaseVolume(){

        if (volumeValue == 10){
            System.out.println("\n> volume is already at maximum: " + volumeValue);
            return;
        }

        this.volumeValue++;
    }

    public void decreaseVolume(){

        if (volumeValue == 1) {
            System.out.println("volume is already at minimum: " + volumeValue);
            return;
        }

        this.volumeValue--;
    }

    public Integer readVolume(){
        return this.volumeValue;
    }

}
